### ISSUE TITLE
Hostapd module - UTF8 SSID

### ISSUE THREAD

**Issue by Jappie3** [NixOS/nixpkgs#286365]
### Describe the bug

I'm messing around with the hostapd module a little & noticed that the value of `utf8_ssid` was my SSID. However, online resources ([1](https://web.mit.edu/freebsd/head/contrib/wpa/hostapd/hostapd.conf) [2](https://github.com/chayleaf/nixos-router/blob/af7d975e755702b649a386c57b44665d56c80d7a/modules/hostapd.nix#L54) [3](https://w1.fi/cgit/hostap/tree/hostapd/config_file.c#n2515)) seem to suggest that this should take a **bool** (0/1) rather than a **string**. This is the link to the current configuration in the hostapd module: https://github.com/NixOS/nixpkgs/blob/25e3d4c0d3591c99929b1ec07883177f6ea70c9d/nixos/modules/services/networking/hostapd.nix#L912 Hostapd doesn't seem to give errors regarding this, but I don't think it is correctly implemented right now.

Could someone else take a quick look at this? I'm fairly inexperienced with hostapd and would love a second opinion / pair of eyes. If I'm correct, the only fix we'd need is changing `utf8_ssid = bssCfg.ssid;` to `utf8_ssid = bssCfg.utf8Ssid;`

### Notify maintainers

@oddlama 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by oddlama** [comment]
You are absolutely right, looking at the initial draft of the module it looks like this has been the victim of a refactor later :P
Thanks for catching this, I've already opened a PR #286393 to address this.
